src/ray/raylet/scheduling/local_task_manager.cc

```diff
@@ -274,10 +274,10 @@ void LocalTaskManager::DispatchScheduledTasksToWorkers() {
       info_by_sched_cls_.erase(scheduling_class);
     }
     if (is_infeasible) {
-      // TODO(scv119): fail the request.
-      // Call CancelTask
-      tasks_to_dispatch_.erase(shapes_it++);
-    } else if (dispatch_queue.empty()) {
+      CancelInfeasiblePlacementGroupTasks(dispatch_queue);
```

Comment:
It will affect the `work_it` of the current scope and make `work_it` invalid.

Reply:
do you mean shape_it?

Reply:
Oh, sorry, I misread it. This line looks good.